> > Current code doesn't do it. The only outcome of it is an error message buried in a usually super-verbose ginkgo output and a 7 minutes(!!!) timeout slowing down the tests.
> 
> I think we should turn this error message into a test failure.

There is missing error checking:
```
ERROR: test/e2e/framework/node/init/init.go:37:27: Error return value of `e2enode.AllNodesReady` is not checked (errcheck)
ERROR: 					e2enode.AllNodesReady(ctx, f.ClientSet, 7*time.Minute)
ERROR: 					                     ^
```

I accidentally dropped the error checking when moving the node support code out of the core framework in 1.26. If we had had today's golangci-lint checks, it would have been flagged as a problem. Because we didn't, it was overlooked... Sorry about that!

--- a/test/e2e/framework/node/init/init.go
+++ b/test/e2e/framework/node/init/init.go
@@ -30,14 +30,10 @@ import (
 func init() {
 	framework.NewFrameworkExtensions = append(framework.NewFrameworkExtensions,
 		func(f *framework.Framework) {
-			ginkgo.AfterEach(func(ctx context.Context) {
-				if f.ClientSet == nil {
-					// Test didn't reach f.BeforeEach, most
-					// likely because the test got
-					// skipped. Nothing to check...
-					return
-				}
-				e2enode.AllNodesReady(ctx, f.ClientSet, 7*time.Minute)
+			ginkgo.BeforeEach(func() {
+				ginkgo.DeferCleanup(func(ctx context.Context) {
+					framework.ExpectNoError(e2enode.AllNodesReady(ctx, f.ClientSet, 7*time.Minute))
+				})
 			})
 		},
 	)

